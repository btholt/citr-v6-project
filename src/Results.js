import Pet from "./Pet";

const Results = ({ pets }) => {
  return (
    <div className="search">
      {!pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => {
          return (
            <Pet
              animal={pet.animal}
              key={pet.id}
              name={pet.name}
              breed={pet.breed}
              images={pet.images}
              location={`${pet.city}, ${pet.state}`}
              id={pet.id}
            />
          );
        })
      )}
    </div>
  );
};

export default Results;
