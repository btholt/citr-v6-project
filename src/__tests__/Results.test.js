import { expect, test } from "@jest/globals";
import { create } from "react-test-renderer";
import Results from "../Results";
import { createRenderer } from "react-test-renderer/shallow";

const pets = [
  {
    id: 1,
    name: "Luna",
    animal: "dog",
    city: "Seattle",
    state: "WA",
    description:
      "Luna is actually the most adorable dog in the world. Her hobbies include yelling at squirrels, aggressively napping on her owners' laps, and asking to be fed two hours before IT'S DAMN WELL TIME LUNA. Luna is beloved by her puppy parents and lazily resides currently in Seattle, Washington.",
    breed: "Havanese",
    images: [
      "http://pets-images.dev-apis.com/pets/dog25.jpg",
      "http://pets-images.dev-apis.com/pets/dog26.jpg",
      "http://pets-images.dev-apis.com/pets/dog27.jpg",
      "http://pets-images.dev-apis.com/pets/dog28.jpg",
      "http://pets-images.dev-apis.com/pets/dog29.jpg"
    ]
  },
  {
    id: 2,
    name: "Charisse",
    animal: "rabbit",
    city: "Lexington",
    state: "KY",
    description:
      "Maecenas leo odio, condimentum id, luctus nec, molestie sed, justo. Pellentesque viverra pede ac diam. Cras pellentesque volutpat dui.\n\nMaecenas tristique, est et tempus semper, est quam pharetra magna, ac consequat metus sapien ut nunc. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Mauris viverra diam vitae quam. Suspendisse potenti.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/rabbit0.jpg"]
  },
  {
    id: 3,
    name: "Maitilde",
    animal: "rabbit",
    city: "Dallas",
    state: "TX",
    description:
      "Duis consequat dui nec nisi volutpat eleifend. Donec ut dolor. Morbi vel lectus in quam fringilla rhoncus.\n\nMauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis. Integer aliquet, massa id lobortis convallis, tortor risus dapibus augue, vel accumsan tellus nisi eu orci. Mauris lacinia sapien quis libero.\n\nNullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit1.jpg"]
  },
  {
    id: 4,
    name: "Natalina",
    animal: "rabbit",
    city: "Tampa",
    state: "FL",
    description:
      "Mauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis. Integer aliquet, massa id lobortis convallis, tortor risus dapibus augue, vel accumsan tellus nisi eu orci. Mauris lacinia sapien quis libero.\n\nNullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.\n\nIn quis justo. Maecenas rhoncus aliquam lacus. Morbi quis tortor id nulla ultrices aliquet.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit2.jpg"]
  },
  {
    id: 5,
    name: "Michail",
    animal: "reptile",
    city: "Tuscaloosa",
    state: "AL",
    description:
      "Nulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/reptile1.jpg"]
  },
  {
    id: 6,
    name: "Gizela",
    animal: "bird",
    city: "Carol Stream",
    state: "IL",
    description:
      "Nullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.\n\nIn quis justo. Maecenas rhoncus aliquam lacus. Morbi quis tortor id nulla ultrices aliquet.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/bird2.jpg"]
  },
  {
    id: 7,
    name: "Laughton",
    animal: "reptile",
    city: "Bridgeport",
    state: "CT",
    description:
      "Cras mi pede, malesuada in, imperdiet et, commodo vulputate, justo. In blandit ultrices enim. Lorem ipsum dolor sit amet, consectetuer adipiscing elit.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/reptile2.jpg"]
  },
  {
    id: 8,
    name: "Si",
    animal: "dog",
    city: "Charlotte",
    state: "NC",
    description:
      "Praesent id massa id nisl venenatis lacinia. Aenean sit amet justo. Morbi ut odio.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/dog0.jpg"]
  },
  {
    id: 9,
    name: "Lyda",
    animal: "rabbit",
    city: "Springfield",
    state: "IL",
    description:
      "Curabitur gravida nisi at nibh. In hac habitasse platea dictumst. Aliquam augue quam, sollicitudin vitae, consectetuer eget, rutrum at, lorem.\n\nInteger tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit3.jpg"]
  },
  {
    id: 10,
    name: "Jackquelin",
    animal: "dog",
    city: "Tucson",
    state: "AZ",
    description:
      "Donec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/dog1.jpg"]
  }
];

// first run will create snapshot file
// future runs will check if the markup has changed
test("snapshot with no pets", () => {
  const tree = create(<Results pets={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// in our snapshot for Results.js, we end up rendering a ton of Pet components
// since we're testing Results, we dont care what Pet looks like
// if test in Results.js fails, we dont want it to be bc of something in Pet.js
// everything should have its own set of tests
// this is why we used shallow render, so it wont actually render Pet, just
// everything going in as props to Pet
test("renders correctly with some pets", () => {
  const r = createRenderer();
  r.render(<Results pets={pets} />);
  expect(r.getRenderOutput()).toMatchSnapshot();
});
