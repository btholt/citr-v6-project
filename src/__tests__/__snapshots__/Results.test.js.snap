// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with some pets 1`] = `
<div
  className="search"
>
  <Pet
    animal="dog"
    breed="Havanese"
    id={1}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/dog25.jpg",
        "http://pets-images.dev-apis.com/pets/dog26.jpg",
        "http://pets-images.dev-apis.com/pets/dog27.jpg",
        "http://pets-images.dev-apis.com/pets/dog28.jpg",
        "http://pets-images.dev-apis.com/pets/dog29.jpg",
      ]
    }
    location="Seattle, WA"
    name="Luna"
  />
  <Pet
    animal="rabbit"
    breed="Havanese"
    id={2}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit0.jpg",
      ]
    }
    location="Lexington, KY"
    name="Charisse"
  />
  <Pet
    animal="rabbit"
    breed="Lab"
    id={3}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit1.jpg",
      ]
    }
    location="Dallas, TX"
    name="Maitilde"
  />
  <Pet
    animal="rabbit"
    breed="Lab"
    id={4}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit2.jpg",
      ]
    }
    location="Tampa, FL"
    name="Natalina"
  />
  <Pet
    animal="reptile"
    breed="Havanese"
    id={5}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/reptile1.jpg",
      ]
    }
    location="Tuscaloosa, AL"
    name="Michail"
  />
  <Pet
    animal="bird"
    breed="Havanese"
    id={6}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/bird2.jpg",
      ]
    }
    location="Carol Stream, IL"
    name="Gizela"
  />
  <Pet
    animal="reptile"
    breed="Havanese"
    id={7}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/reptile2.jpg",
      ]
    }
    location="Bridgeport, CT"
    name="Laughton"
  />
  <Pet
    animal="dog"
    breed="Lab"
    id={8}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/dog0.jpg",
      ]
    }
    location="Charlotte, NC"
    name="Si"
  />
  <Pet
    animal="rabbit"
    breed="Lab"
    id={9}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit3.jpg",
      ]
    }
    location="Springfield, IL"
    name="Lyda"
  />
  <Pet
    animal="dog"
    breed="Lab"
    id={10}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/dog1.jpg",
      ]
    }
    location="Tucson, AZ"
    name="Jackquelin"
  />
</div>
`;

exports[`snapshot with no pets 1`] = `
<div
  className="search"
>
  <h1>
    No Pets Found
  </h1>
</div>
`;
